src/main/java/net/pms/dlna/DLNAMediaAudio.java

```diff
@@ -509,40 +509,63 @@ public String getAudioCodec() {
 	@Override
 	public String toString() {
 		StringBuilder result = new StringBuilder();
-		result.append("id: ");
-		result.append(getId());
-		result.append(", lang: ");
-		result.append(getLang());
+		if (!getLang().equals("und")) {
+			result.append("Id: ").append(getId());
+			result.append(", Language Code: ").append(getLang());
+		}
 
 		if (isNotBlank(getAudioTrackTitleFromMetadata())) {
-			result.append(", audio track title from metadata: ");
-			result.append(getAudioTrackTitleFromMetadata());
+			if (result.length() > 0) {
+				result.append(", ");
+			}
+			result.append("Audio Track Title From Metadata: ").append(getAudioTrackTitleFromMetadata());
 		}
 
-		result.append(", audio codec: ");
-		result.append(getAudioCodec());
-		result.append(", sample frequency:");
-		result.append(getSampleFrequency());
-
-		if (getAudioProperties() != null && getAudioProperties().getNumberOfChannels() != 0) {
-			result.append(", number of channels: ");
-			result.append(getAudioProperties().getNumberOfChannels());
+		if (result.length() > 0) {
+			result.append(", ");
+		}
+		result.append("Audio Codec: ").append(getAudioCodec());
+		if (getSampleFrequency() != null) {
+			result.append(", Sample Frequency: ").append(getSampleFrequency());
+		}
+		if (getBitRate() != 0) {
+			result.append(", Bitrate: ").append(getBitRate());
+		}
+		if (getBitsperSample() != 0 && getBitsperSample() != 16 ) {
+			result.append(", Bits per Sample: ").append(getBitsperSample());
 		}
 
-		result.append(", bits per sample: ");
-		result.append(getBitsperSample());
+		if (getAudioProperties() != null) {
+			result.append(", ").append(getAudioProperties());
```

Comment:
Again, this comes down to how ```toString()``` works and what it's supposed to do. You only thinking of one specific situation where you are looking for certain information in the log. That's not what it's for, it is also used in other circumstances like while debugging. ```DLNAMediaAudio``` and ```AudioProperties``` need to be able to represent themselves also, they are not always "used together". You must think of ```AudioProperites``` as an object instance, and ```DLNAMediaAudio``` as another. One ```AudioProperties``` instance can be "linked" to ```DLNAMediaAudio```, in which case we choose to print the content of ```AudioProperties``` together with ```DLNAMediaAudio``` - but ```AudioProperites``` still needs to be able to be presented alone. With the same logic, hiding information in ```DLNAMediaInfo``` because this information also exists in ```AudioProperties``` is wrong - the information is still existing in ```DLNAMediaInfo``` and although they might be the same most of the time (or always), you're not actually printing the same information. They can be different, and if they are we probably have a problem.